ipatests/test_integration/test_idp.py

```diff
@@ -58,11 +62,11 @@ def kinit_idp(host, user, keycloak_server):
     tasks.kdestroy_all(host)
     # create armor for FAST
     host.run_command(["kinit", "-n", "-c", ARMOR])
-    since = time.strftime('%Y-%m-%d %H:%M:%S')
     cmd = ["kinit", "-T", ARMOR, user]
     with host.spawn_expect(cmd, default_timeout=100) as e:
         e.expect('Authenticate at .+: ')
-        uri = get_verification_uri(host, since, keycloak_server.hostname)
+        time.sleep(15)
+        uri = get_verification_uri(host, keycloak_server.hostname)
```

Comment:
The `get_verification_uri` seems a complex way of retrieving the URI, by reading the content of the journal.
The code just above, `e.expect('Authenticate at .+: ')` could probably be  used to retrieve the URI, as the output is similar to `Authenticate at https://replica1.ipa.test:8443/auth/realms/master/device?user_code=OWAM-ZOPR and press ENTER.: `. I believe it's possible to extract the url directly from the output using e.match.groups() if you change e.expect to something like e.expect(Authenticate at (.+) and press ENTER.:`)